src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs

```diff
@@ -123,34 +123,80 @@ SyntaxNodeAnalysisContext ctx
 				);
 
 				return;
-			}
-
-			// Literal arguments should always be named
-			foreach( var arg in unnamedArgs ) {
-
-				// Check if the argument type is literal
-				if( arg.Syntax.Expression is LiteralExpressionSyntax ) {
-
+			}
+
+			int numOfSwappableArgs = 0;
+			foreach( var arg in unnamedArgs ) {
+
+				// Literal expressions have their own issue
+				if( arg.Syntax.Expression is LiteralExpressionSyntax ) {
+					// Literal arguments should always be named
 					var fixerContext = CreateFixerContext( ImmutableArray.Create( arg ) );
-
 					ctx.ReportDiagnostic( Diagnostic.Create(
 							descriptor: Diagnostics.LiteralArgShouldBeNamed,
 							location: arg.Syntax.Expression.GetLocation(),
 							properties: fixerContext,
-							messageArgs: arg.ParamName
-						)
-					);
+							messageArgs: arg.ParamName ) );
+					continue;
+				}
+
+				// Take note of the type of argument
+				ITypeSymbol argType = GetArgumentType( ctx, arg.Syntax );
+
+				// Iterate through the other arguments in the list
+				foreach( var compareArg in unnamedArgs ) {
+					if( compareArg.Position == arg.Position ) {
+						continue;
+					}
+
+					// Get the associated parameter of the argument to compare
+					IParameterSymbol param = compareArg.Syntax.DetermineParameter(
+						ctx.SemanticModel,
+						allowParams: false );
+
+					// Check if the argument could apply to the other parameter
+					if( ctx.SemanticModel.Compilation.ClassifyConversion(
+						argType,
+						param.Type ).Exists ) {
+						// The argument can apply to multiple parameters,
+						// so take note of it
+						numOfSwappableArgs++;

```

Comment:
A few thoughts:

* I think we are complaining here if an argument could be used in the place of another parameter, but we actually care if two arguments could be swapped. So like with `Foo( object x, int x )` we aren't worried about `Foo( new object(), 6 )` even though `6` could go in place of the first parameter.
* Do we want to output a diagnostic for just these ambiguous arguments, or just say "screw it, name every argument"? I think we want to do it individually for these (like we do for literals). We'd need to keep track of the indexes of arguments that are (mutually) swappable, like a `List<int>` or something?
* Actually, now I wonder if we should just be building up a single diagnostic... like instead of immediately emitting a diagnostic for literals we should be adding to a shared `fixerContext` instance and keeping a `bool hadProblems` to emit one diagnostic at the end.

For context: the fixer takes possibly multiple arguments to name in the diagnostic. Ideally I think we'd have at most one diagnostic per function call that we are checking.

Reply:
Arguments are now flagged and reported individually. Your example is now handled.